internal/policy/targets_test.go

```diff
@@ -86,6 +86,75 @@ func TestUpdateDelegation(t *testing.T) {
 	}, targetsMetadata.Delegations.Roles[0])
 }
 
+func TestReorderDelegations(t *testing.T) {
+	targetsMetadata := InitializeTargetsMetadata()
+
+	key1, err := tuf.LoadKeyFromBytes(targets1PubKeyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+	key2, err := tuf.LoadKeyFromBytes(targets2PubKeyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	targetsMetadata, err = AddDelegation(targetsMetadata, "rule-1", []*tuf.Key{key1}, []string{"path1/"}, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	targetsMetadata, err = AddDelegation(targetsMetadata, "rule-2", []*tuf.Key{key2}, []string{"path2/"}, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	targetsMetadata, err = AddDelegation(targetsMetadata, "rule-3", []*tuf.Key{key1, key2}, []string{"path3/"}, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := map[string]struct {
+		ruleNames     []string
+		expected      []string
+		expectedError error
+	}{
+		"reverse order (valid input)": {
+			ruleNames:     []string{"rule-3", "rule-2", "rule-1"},
+			expected:      []string{"rule-3", "rule-2", "rule-1", AllowRuleName},
+			expectedError: nil,
+		},
+		"rule not specified in new order": {
+			ruleNames:     []string{"rule-3", "rule-2"},
+			expectedError: ErrMissingRules,
+		},
+		"rule repeated in the new order": {
+			ruleNames:     []string{"rule-3", "rule-2", "rule-1", "rule-3"},
+			expectedError: ErrDuplicatedRuleName,
+		},
+		"unknown rule in the new order": {
```

Comment:
as noted above, this should also check when the specified list is the right length. Without this, we can panic when looking up a rule in rolesMap.